includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php

```diff
@@ -1206,7 +1206,7 @@ public function add_contact( $contact, $list_id = false ) {
 			// Log the error with any details returned from the API.
 			do_action(
 				'newspack_log',
-				'newspack_constant_contact_add_contact_failed',
+				'newspack_' . $this->service . '_api_error',
```

Comment:
this logging repeats in every esp. Is it possible to move this up in the abstraction and automatically do it for all of them?

Also, wdyt of adding all errors into the same bucket and add the provider as a piece of data in the params. It's what we do with all the other logs

Reply:
4acbef7 implements this change. It simplifies things a bit by repurposing the existing log with the `newspack_esp_sync_upsert_contact` error code to log the error message from the ESP's API instead of the reader-facing error message, so there's now no longer a separate log for each. There's probably no reason to log the reader-facing error message for debugging purposes.

The testing instructions remain the same, but you can also verify that the logged output shows the original error message from the API instead of the reader-facing error message:

<img width="806" alt="Screenshot 2025-02-18 at 12 35 17 PM" src="https://github.com/user-attachments/assets/e1588992-c450-47c7-b5e3-8c7ffa707372" />